nncf/experimental/torch/fx/transformations.py

```diff
@@ -136,17 +142,47 @@ def bias_update_transformation(model: torch.fx.GraphModule):
     return bias_update_transformation
 
 
-def constant_update_transformation_builder(node: NNCFNode, value: torch.Tensor) -> TransformationFNType:
+def shared_constant_create_transformation(model: torch.fx.GraphModule):
+    """
+    Return transformation which checks fx graph for shared constants, disconnects
+    and eliminates redundant shared constant while connecting singular shared constant.
+    :return: Transformation which attaches shared constants to nodes and removes redundant constants.
+    """
+    prev_targets = {}
+
+    for source_node in model.graph.nodes:
+        _replace_shared_weights(source_node, prev_targets)
+
+    model.graph.eliminate_dead_code()
+    model.recompile()
+
+
+def _replace_shared_weights(node: torch.fx.Node, prev_targets):
+    """
+    This function is responsible for checking the consumer node of current
```

Comment:
Please do not create an inner function and put this function body to the `shared_constant_create_transformation`

Reply:
Done